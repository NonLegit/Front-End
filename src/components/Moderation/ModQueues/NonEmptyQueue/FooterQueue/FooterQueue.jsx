// import ShortcutOutlinedIcon from '@mui/icons-material/ShortcutOutlined';
import CheckIcon from '@mui/icons-material/Check';
import CloseIcon from '@mui/icons-material/Close';
import {
  Divider, Menu, MenuItem, Typography,
} from '@mui/material';
import { useState } from 'react';
import MoreHorizOutlinedIcon from '@mui/icons-material/MoreHorizOutlined';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
// import ShortcutOutlinedIcon from '@mui/icons-material/ShortcutOutlined';
import VisibilityOffOutlinedIcon from '@mui/icons-material/VisibilityOffOutlined';
import EighteenUpRatingOutlinedIcon from '@mui/icons-material/EighteenUpRatingOutlined';
import LockIcon from '@mui/icons-material/Lock';
// import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
// import AdminPanelSettingsIcon from '@mui/icons-material/AdminPanelSettings';
// import AdminPanelSettingsOutlinedIcon from '@mui/icons-material/AdminPanelSettingsOutlined';
// import MoreHorizOutlinedIcon from '@mui/icons-material/MoreHorizOutlined';
import CancelPresentationOutlinedIcon from '@mui/icons-material/CancelPresentationOutlined';
import {
  FooterBox, FooterButton, MoreList, SelectItem,
} from './styles';

/**
 * left side bar on top communities
 *
 * @component Categories
 * @property {string} modState - the state of the post (approved-removed-spam)
 * @property {boolean} locked - is the post locked
 * @property {boolean} hidden - is the post hidden
 * @property {function} handleHide - function to handle hide event
 * @property {function} handleLock - function to handle lock comments event
 * @property {function} handleNsfw - function to handle mark as nsfw event
 * @property {function} handleApprove - function to handle approve event
 * @property {function} handleRemove - function to handle remove event
 * @property {function} handleSpam - function to handle spam event
 * @returns {React.Component} Categories
 */
function FooterQueue(props) {
  const {
    modState,
    locked,
    hidden,
    handleHide,
    handleLock, handleNsfw, handleApprove, handleRemove, handleSpam,
  } = props;

  const [anchorEl, setAnchorEl] = useState(null);
  const open = Boolean(anchorEl);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  // const handleShare = () => {
  //   console.log('share');
  // };

  return (
    <FooterBox>
      {modState !== 'approved'
        ? (
          <FooterButton variant="contained" startIcon={<CheckIcon fontSize="small" />} onClick={handleApprove}>Approve</FooterButton>
        )
        : <FooterButton variant="outlined" startIcon={<CheckIcon fontSize="medium" />} condition="true" onClick={handleApprove}>Approved</FooterButton>}
      <FooterButton variant="outlined" startIcon={<CloseIcon fontSize="medium" />} condition="true" onClick={handleRemove}>Remove</FooterButton>
      <MoreList onClick={handleClick}>
        <MoreHorizOutlinedIcon />
      </MoreList>
      <Menu
        id="long-menu"
        MenuListProps={{
          'aria-labelledby': 'long-button',
        }}
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        PaperProps={{
          style: {
            width: '170px',
            padding: 5,
          },
        }}
        sx={{ '& .MuiButtonBase-root': { padding: '3px 3px' } }}
      >
        <Typography
          variant="subtitle2"
          sx={{
            fontWeight: 700, color: '#4b6066', marginLeft: 1, fontSize: '12px',
          }}
        >
          MODERATION
        </Typography>

        <MenuItem>
          <SelectItem onClick={handleSpam}>
            <CancelPresentationOutlinedIcon sx={{ marginRight: 1 }} fontSize="small" />
            Remove As Spam
          </SelectItem>

        </MenuItem>
        {/* <Divider /> */}

        <MenuItem>

          <SelectItem onClick={handleLock}>
            {!locked ? <LockOutlinedIcon sx={{ marginRight: 1 }} fontSize="small" /> : <LockIcon sx={{ marginRight: 1 }} fontSize="small" />}
            Lock Comments
          </SelectItem>

        </MenuItem>
        {/* <Divider /> */}
        <MenuItem>

          <SelectItem onClick={handleNsfw}>
            <EighteenUpRatingOutlinedIcon sx={{ marginRight: 1 }} fontSize="small" />
            Mark As NSFW
          </SelectItem>

        </MenuItem>
        <Divider />

        <Typography
          variant="subtitle2"
          sx={{
            fontWeight: 700, color: '#4b6066', marginLeft: 1, fontSize: '12px',
          }}
        >
          OTHER
        </Typography>

        {/* <MenuItem>

          <SelectItem onClick={handleShare}>
            <ShortcutOutlinedIcon sx={{ marginRight: 1 }} fontSize="small" />
            Share
          </SelectItem>

        </MenuItem> */}

        <MenuItem>
          <SelectItem onClick={handleHide}>
            <VisibilityOffOutlinedIcon sx={{ marginRight: 1 }} fontSize="small" />
            {hidden ? 'Unhide' : 'Hide'}
          </SelectItem>
        </MenuItem>

      </Menu>
    </FooterBox>
  );
}

export default FooterQueue;
