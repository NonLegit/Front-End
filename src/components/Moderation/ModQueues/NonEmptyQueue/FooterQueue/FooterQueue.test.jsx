import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import FooterQueue from './FooterQueue';

test('FooterQueue test snapshot', async () => {
  const tree = renderer.create(
    <FooterQueue />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
