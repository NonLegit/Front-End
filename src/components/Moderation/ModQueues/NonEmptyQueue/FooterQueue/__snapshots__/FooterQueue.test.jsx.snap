// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterQueue test snapshot 1`] = `
<div
  className="MuiBox-root css-1lye4r5"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1kyghbm-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="CheckIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
        />
      </svg>
    </span>
    Approve
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-te5573-MuiButtonBase-root-MuiButton-root"
    condition="true"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </span>
    Remove
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1mcolru-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MoreHorizOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </button>
</div>
`;
